src/software/er_force_simulator_main.cpp

```diff
@@ -121,12 +122,24 @@ int main(int argc, char **argv)
         auto simulator_state_output = ThreadedProtoUnixSender<world::SimulatorState>(
             runtime_dir + SIMULATOR_STATE_PATH);
 
+        // Simulation Started Trigger as Simulator Output
+        auto simulation_started_trigger =
+            ThreadedProtoUnixSender<TbotsProto::SimulationStartedTrigger>(
+                runtime_dir + SIMULATION_STARTED_TRIGGER_PATH);
+
         // Inputs
         // World State Input: Configures the ERForceSimulator
         auto world_state_input = ThreadedProtoUnixListener<TbotsProto::WorldState>(
             runtime_dir + WORLD_STATE_PATH, [&](TbotsProto::WorldState input) {
                 std::scoped_lock lock(simulator_mutex);
                 er_force_sim->setWorldState(input);
+
+                auto sim_started_trigger_msg = *createSimulationStartedTrigger(true);
+                if (!er_force_sim->has_sent_sim_start_trigger)
+                {
+                    simulation_started_trigger.sendProto(sim_started_trigger_msg);
+                    er_force_sim->has_sent_sim_start_trigger = true;
+                }
```

Comment:
So right now you're sending the `SimulationStartedTrigger` inside of this [lambda](https://www.programiz.com/cpp-programming/lambda-expression) function. This function is a callback for when a `WorldState` protobuf is received from Python. We know that we don't want to send the `WorldState` protobuf message _until_ the we know for a fact that the simulation is setup and ready to receive it (i.e. `SimulationStartedTrigger` has been sent from C++). So both processes are basically waiting for the other process to send the protobuf message, making a deadlock.

What we should do is move this part of the code (specifically sending the proto) out of this lambda function. IMO it would make the most sense to send the simulated started trigger message, once the simulation has started (in other words, the `ErForceSimulator` object + the callbacks in this file have been fully setup. That would mean that you can add this code to the end of this file (but before `std::promise<void>().get_future().wait();` since will block and no code after it will be executed)